cmd/dstask.go

```diff
@@ -265,34 +265,80 @@ func main() {
 		}
 
 	case dstask.CMD_EDIT:
-		ts := dstask.LoadTasksFromDisk(dstask.NON_RESOLVED_STATUSES)
-		for _, id := range cmdLine.IDs {
-			task := ts.MustGetByID(id)
+		// If UUID present, load resolved tasks. If not load non-resolved tasks.
+		if cmdLine.UUID != "" {
+			ts := dstask.LoadTasksFromDisk([]string{dstask.STATUS_RESOLVED})
+			// Search for UUIDs that match uuid: string
+			var uuids []string = ts.SearchForUUID(cmdLine.UUID)
+
+			if len(uuids) == 0 { // No matches
+				fmt.Printf("No Resolved Tasks Match UUID: %s\n", cmdLine.UUID)
+
+			} else if len(uuids) > 1 { // More than one UUID found that matches string
+
+				fmt.Printf("Multiple Tasks match UUID: %s\n", cmdLine.UUID)
+				// Print UUIDs and Summary and prompt user to enter more specific ID.
+				// Probably better to either filter ts or create new taskset with
+				// uuids then can use DisplayByNext
+				fmt.Printf("\nUUID \t\t\t\t\tSummary\n")
+				for _, uuid := range uuids {
+					task := ts.MustGetByUUID(uuid)
+					fmt.Printf("%s \t%s\n", task.UUID, task.Summary)
+				}
+				fmt.Println("\nProvide provide more specific UUID")
+
+			} else { // Edit resolved task
+				task := ts.MustGetByUUID(uuids[0])
+				data, err := yaml.Marshal(&task)
+				if err != nil {
+					// TODO present error to user, specific error message is important
+					dstask.ExitFail("Failed to marshal task %s", task)
+				}
 
-			// hide ID
-			task.ID = 0
+				data = dstask.MustEditBytes(data, "yml")
 
-			data, err := yaml.Marshal(&task)
-			if err != nil {
-				// TODO present error to user, specific error message is important
-				dstask.ExitFail("Failed to marshal task %s", task)
+				err = yaml.Unmarshal(data, &task)
+				if err != nil {
+					// TODO present error to user, specific error message is important
+					// TODO reattempt mechanism
+					dstask.ExitFail("Failed to unmarshal yml")
```

Comment:
```suggestion
					dstask.ExitFail(fmt.Sprintf("Failed to unmarshal yml: %v", err))
```
If this error does happen, it would probably mean a corrupted file or a manual edit. This is one way we could show the underlying error.

Reply:
Yes, that could be helpful. We could utilise [ConfirmOrAbort](https://github.com/naggie/dstask/blob/master/util.go#L20) in a loop to enable the user to correct mistakes, too (another one I had planned but never got round to)